File: backend/app/schemas/cover_letter/cover_letter.py
from datetime import datetime
from typing import Any, Dict

from pydantic import BaseModel


class CoverLetterBase(BaseModel):
    title: str

    template: str

    data: Dict[str, Any]

    latest_cover_letter_analysis: dict | None = None


class CoverLetterCreate(CoverLetterBase):
    pass


class CoverLetterResponse(CoverLetterBase):
    id: int

    created_at: datetime

    updated_at: datetime

    latest_cover_letter_analysis_created_at: datetime | None = None

    class Config:
        from_attributes = True

File: frontend/src/features/cover-letter/types/coverLetter.ts
export type CoverLetterTemplate =
  | "classic"
  | "modern"
  | "minimal"
  | "startup";

export interface CoverLetterTypography {
  fontFamily: string;

  fontSize: number;

  lineHeight: number;
}

export interface CoverLetterSender {
  fullName: string;

  currentRole?: string;

  skills?: string[];

  street: string;

  zip: string;

  city: string;

  email: string;

  phone: string;
}

export interface CoverLetterRecipient {
  companyName: string;

  contactName: string;

  street: string;

  zip: string;

  city: string;
}

export interface CoverLetterContent {
  subject: string;

  body: string;

  closing: string;
}

export interface CoverLetter {
  id?: number | string;

  title: string;

  selectedResumeId?: string | number;

  resumeContext?: string;

  structuredResumeData?: Record<string, any> | null;

  language?: "english" | "german";

  jobPosting: string;

  tone: string;

  template: CoverLetterTemplate;

  typography?: CoverLetterTypography;

  sender: CoverLetterSender;

  recipient: CoverLetterRecipient;

  date: string;

  createdAt?: string;

  updatedAt?: string;

  created_at?: string;

  updated_at?: string;

  latest_cover_letter_analysis?: Record<string, any> | null;

  latest_cover_letter_analysis_created_at?: string | null;

  content: CoverLetterContent;
}